Index: org.glite.deployment.lb/config/scripts/glite-lb-config.py
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.lb/config/scripts/glite-lb-config.py,v
retrieving revision 1.65.2.3
diff -u -r1.65.2.3 glite-lb-config.py
--- org.glite.deployment.lb/config/scripts/glite-lb-config.py	2 May 2006 10:36:16 -0000	1.65.2.3
+++ org.glite.deployment.lb/config/scripts/glite-lb-config.py	30 Jun 2006 09:29:21 -0000
@@ -127,7 +127,12 @@
 
         if not os.path.exists('/tmp/mysql.sock'):
             os.symlink('/var/lib/mysql/mysql.sock', '/tmp/mysql.sock')
-            
+        
+
+        #-------------------------------------------------------------------
+        # start bkserver
+        #-------------------------------------------------------------------
+
         pid = glib.getPID('bkserverd')
         if pid != 0:
             print 'The gLite LB Server service is already running. Restarting...'
@@ -140,7 +145,7 @@
         pid = glib.getPID('bkserverd')
 
         if (pid != 0):
-            print "The gLite LB Server service has been started               ",
+            print "The gLite LB Server service has been started              ",
             glib.printOkMessage()
         else:
             glib.printErrorMessage("Could not start the gLite LB Server service")
@@ -149,6 +154,36 @@
             return 1
         
         #-------------------------------------------------------------------
+        # start jp-importer, if enabled
+        #-------------------------------------------------------------------
+
+        lb_export = 0
+        if params.has_key('lb.export.enabled'):
+            if params['lb.export.enabled'] == "true":
+                lb_export = 1
+
+        if lb_export:
+            pid = glib.getPID('jp-importer')
+            if pid != 0:
+                print 'The gLite JP Importer service is already running. Restarting...'
+                os.system('%s/etc/init.d/glite-jp-importer stop' % os.environ['GLITE_LOCATION'])
+            else:
+                print 'Starting the gLite JP Importer service...'
+
+            os.system('%s/etc/init.d/glite-jp-importer start' % os.environ['GLITE_LOCATION'])
+
+            pid = glib.getPID('jp-importer')
+
+            if (pid != 0):
+                print "The gLite JP Importer service has been started            ",
+                glib.printOkMessage()
+            else:
+                glib.printErrorMessage("Could not start the gLite JP Importer service")
+                glib.printErrorMessage("Please verify and re-run the script                        "),
+                glib.printFailedMessage()
+                return 1
+        
+        #-------------------------------------------------------------------
         # Start Servicetool
         #-------------------------------------------------------------------
 
@@ -168,6 +203,10 @@
         if (pid != 0):
             os.system('%s/etc/init.d/glite-lb-bkserverd stop' % os.environ['GLITE_LOCATION'])
 
+        #-------------------------------------------------------------------
+        # Book Keeping Server
+        #-------------------------------------------------------------------
+
         pid = glib.getPID('bkserverd')
         if (pid != 0):
             print 'Could not stop the LB Server service            ',
@@ -178,6 +217,25 @@
             glib.printOkMessage()
         
         #-------------------------------------------------------------------
+        # JP Importer
+        #-------------------------------------------------------------------
+
+        pid = glib.getPID('jp-importer')
+        if (pid != 0):
+            os.system('%s/etc/init.d/glite-jp-importer stop' % os.environ['GLITE_LOCATION'])
+
+        pid = glib.getPID('jp-importer')
+        if (pid != 0):
+            print 'Could not stop the JP Importer service                  ',
+            glib.printFailedMessage()
+            error_level = 1
+        else:
+            if params.has_key('lb.export.enabled'):
+                if params['lb.export.enabled'] == "true":
+                    print 'The JP Importer service has been stopped                  ',
+                    glib.printOkMessage()
+        
+        #-------------------------------------------------------------------
         # MySQL
         #-------------------------------------------------------------------
 
@@ -202,6 +260,10 @@
         if retval != 0:
             error_level = 1
 
+        retval = os.system('%s/etc/init.d/glite-jp-importer status' % os.environ['GLITE_LOCATION'])
+        if retval != 0:
+            error_level = 1
+
         #-------------------------------------------------------------------
         # Servicetool
         #-------------------------------------------------------------------
@@ -332,7 +394,27 @@
             print "\n==> MySQL database %s already exist\n" % params['lb.database.name']
             
         self.mysql.stop()
-    
+
+        # ------------------------------------------------------------------
+        # export from bkserver to cron
+        # ------------------------------------------------------------------
+	if params['GLITE_LB_EXPORT_ENABLED'] == "true":
+	    file = open('%s/etc/glite-lb-export-cron-wrapper.sh' % os.environ['GLITE_LOCATION'], 'w')
+	    file.write('#! /bin/sh\n')
+	    file.write('. %s\n' % glib.getInstallerExportFile())
+            file.write('$GLITE_LOCATION/examples/glite-lb-export.sh\n')
+	    file.close()
+            os.system('/bin/chmod 0755 %s/etc/glite-lb-export-cron-wrapper.sh' % os.environ['GLITE_LOCATION'])
+
+	    file = open('/etc/cron.d/glite-lb-export.cron', 'w')
+	    file.write('# periodically run purge and export jobs from bkserver\n')
+	    file.write('00,30 * * * * %s %s/etc/glite-lb-export-cron-wrapper.sh\n' % (params['glite.user.name'], os.environ['GLITE_LOCATION']))
+	    file.close()
+            os.system('/bin/chmod 0755 /etc/cron.d/glite-lb-export.cron')
+
+            # Touch cron spool directory to cause reloading of the crontabs
+            os.system("/bin/touch /var/spool/cron")
+
         #-------------------------------------------------------------------
         # RGMA servicetool: configure servicetool
         #-------------------------------------------------------------------
@@ -418,6 +500,17 @@
     # Perl
     glib.addEnvPath("PERL5LIB", "%s/lib/perl:%s/lib/perl5" % (os.environ['GPT_LOCATION'],os.environ['GLITE_LOCATION']))
 
+    # LB export
+    glib.export('GLITE_LB_EXPORT_ENABLED', params['lb.export.enabled']);
+    glib.export('GLITE_LB_EXPORT_BKSERVER', params['lb.export.bkserver']);
+    glib.export('GLITE_LB_EXPORT_JPPS', params['lb.export.jpps']);
+    glib.export('GLITE_LB_EXPORT_JPREG_MAILDIR', params['lb.export.jpreg']);
+    glib.export('GLITE_LB_EXPORT_JPDUMP_MAILDIR', params['lb.export.jpdump']);
+    glib.export('GLITE_LB_EXPORT_DUMPDIR', params['lb.export.dump']);
+    glib.export('GLITE_LB_EXPORT_DUMPDIR_OLD', params['lb.export.dump.old']);
+    glib.export('GLITE_LB_EXPORT_EXPORTDIR', params['lb.export.export']);
+    glib.export('GLITE_LB_EXPORT_PURGE_ARGS', '"%s"' % params['lb.export.purgeargs']);
+
     # Set environment
     glib.setUserEnv()
     
Index: org.glite.deployment.lb/config/templates/glite-lb.cfg.xml
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.lb/config/templates/glite-lb.cfg.xml,v
retrieving revision 1.21
diff -u -r1.21 glite-lb.cfg.xml
--- org.glite.deployment.lb/config/templates/glite-lb.cfg.xml	13 Mar 2006 15:19:21 -0000	1.21
+++ org.glite.deployment.lb/config/templates/glite-lb.cfg.xml	30 Jun 2006 09:29:21 -0000
@@ -17,9 +17,16 @@
 			parameter. Leave it empty of comment it out to use the same as 'glite.user.name'"
 			value="changeme"/>
              	
-	    <mysql.root.password
-	        description="The mysql root password"
-	        value="changeme"/>
+		<mysql.root.password
+		        description="The mysql root password"
+	        	value="changeme"/>
+
+		<lb.export.jpps
+			description="Job Provenance Primary Storage.
+				[Example: localhost:8901][Type: string]"
+			value="changeme"/>
+
+
 	
 		<!-- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx -->
 		<!-- Advanced parameters - Change them if you know what you're doing -->
@@ -55,6 +62,12 @@
 				[Example: 17M][Type: Integer][Unit: MB]"
 			value="17M"/>
 
+		<lb.export.enabled
+			description="Enables exports to Job Provenance.
+				[Example: true][Type: boolean]"
+			value="true"/>
+
+
 		<!-- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx -->
 		<!-- System parameters - You should leave these alone -->
 		<!-- xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx -->		
@@ -67,7 +80,42 @@
 			<value>destination</value>
 
 		</lb.index.list>
-			
+
+		<lb.export.bkserver
+			description="Book Keeping Server service when differs from default port on localhost.
+				[Example: localhost:9000][Type: string]"
+			value=""/>
+
+		<lb.export.jpreg
+			description="Maildir for job registrations.
+				[Example: /tmp/jpreg][Type: string]"
+			value=""/>
+
+		<lb.export.jpdump
+			description="Maildir for job dumps.
+				[Example: /tmp/jpdump][Type: string]"
+			value=""/>
+
+		<lb.export.dump
+			description="Dump directory for purged jobs.
+				[Example: /tmp/dump][Type: string]"
+			value=""/>
+
+		<lb.export.dump.old
+			description="Dump directory for handled purged jobs.
+				[Example: /tmp/dump.old][Type: string]"
+			value=""/>
+
+		<lb.export.export
+			description="Exported events divided to file per job.
+				[Example: /tmp/lb_export][Type: string]"
+			value=""/>
+
+		<lb.export.purgeargs
+			description="Purge arguments (with timeouts).
+				[Example: -a 1h -c 1h -n 1h -o 1d][Type: string]"
+			value="-a 1d -c 1d -n 1d -o 7d"/>
+
 	</parameters>
 
 </config>
Index: org.glite.deployment.lb/project/glite-lb.sdf.xml.template
===================================================================
RCS file: /cvs/jra1mw/org.glite.deployment.lb/project/glite-lb.sdf.xml.template,v
retrieving revision 1.38.2.2
diff -u -r1.38.2.2 glite-lb.sdf.xml.template
--- org.glite.deployment.lb/project/glite-lb.sdf.xml.template	2 May 2006 10:36:19 -0000	1.38.2.2
+++ org.glite.deployment.lb/project/glite-lb.sdf.xml.template	30 Jun 2006 09:29:21 -0000
@@ -57,6 +57,12 @@
 					build="@org.glite.lb.server-bones.info.build@"
 					arch="i386"/>
 					   
+				<component name="glite-lb-client"
+					version="@org.glite.lb.client.info.version@"
+					age="@org.glite.lb.client.info.age@"
+					build="@org.glite.lb.client.info.build@"
+					arch="i386"/>
+					   
 				<component name="glite-wms-utils-jobid"
 					version="@org.glite.wms-utils.jobid.info.version@"
 					age="@org.glite.wms-utils.jobid.info.age@"
@@ -105,6 +111,12 @@
                     build="@org.glite.jp.ws-interface.info.build@"
                     arch="i386"/>
 
+				<component name="glite-jp-client"
+                    version="@org.glite.jp.client.info.version@"
+                    age="@org.glite.jp.client.info.age@"
+                    build="@org.glite.jp.client.info.build@"
+                    arch="i386"/>
+
                 <component name="gridsite"
                     version="@org.gridsite.core.info.version@"
                     age="@org.gridsite.core.info.age@"
