Index: org.gridsite.core/src/Makefile
===================================================================
RCS file: /cvs/jra1mw/org.gridsite.core/src/Makefile,v
retrieving revision 1.48
diff -u -r1.48 Makefile
--- org.gridsite.core/src/Makefile	18 Apr 2006 10:59:01 -0000	1.48
+++ org.gridsite.core/src/Makefile	21 Apr 2006 14:14:53 -0000
@@ -56,9 +56,9 @@
 # Build
 #
 
-build: apidoc \
-       libgridsite.so.$(VERSION) libgridsite.a htcp mod_gridsite.so \
-       urlencode findproxyfile real-gridsite-admin.cgi gsexec \
+build: \
+       libgridsite.so.$(VERSION) libgridsite.a htcp \
+       urlencode findproxyfile real-gridsite-admin.cgi \
        gridsite-copy.cgi
 
 build: libgridsite_globus.so.$(VERSION) libgridsite_globus.a
@@ -226,7 +226,7 @@
 # Install
 #
 
-install: apidoc
+install: 
 	mkdir -p $(prefix)/include \
                  $(prefix)/lib \
                  $(prefix)/bin \
@@ -273,8 +273,6 @@
 	ln -sf htcp $(prefix)/bin/htmv
 	ln -sf htcp $(prefix)/bin/htping
 	ln -sf htcp $(prefix)/bin/htfind
-	cp -f gsexec $(prefix)/sbin
-	cp -f mod_gridsite.so $(prefix)/lib/httpd/modules
 
 #
 # Distributions
