Index: org.gridsite.core/src/Makefile
===================================================================
RCS file: /cvs/jra1mw/org.gridsite.core/src/Makefile,v
retrieving revision 1.53
diff -u -r1.53 Makefile
--- org.gridsite.core/src/Makefile	31 May 2006 11:25:33 -0000	1.53
+++ org.gridsite.core/src/Makefile	8 Jun 2006 15:42:50 -0000
@@ -61,7 +61,7 @@
        urlencode findproxyfile real-gridsite-admin.cgi gsexec \
        gridsite-copy.cgi
 
-build: libgridsite_globus.so.$(VERSION) libgridsite_globus.a
+build-lib: libgridsite_globus.so.$(VERSION) libgridsite_globus.a
 
 # First, normal versions using system OpenSSL rather than Globus OpenSSL
 
@@ -233,7 +233,8 @@
 # Install
 #
 
-install: apidoc
+
+install: apidoc install-lib
 	mkdir -p $(prefix)/include \
                  $(prefix)/lib \
                  $(prefix)/bin \
@@ -248,22 +249,6 @@
 	cp -f findproxyfile $(prefix)/bin
 	cp -f real-gridsite-admin.cgi $(prefix)/sbin
 	cp -f gridsite-copy.cgi $(prefix)/sbin
-	cp -f  libgridsite.a $(prefix)/lib
-	cp -f  libgridsite.so.$(PATCH_VERSION) $(prefix)/lib
-	ln -sf libgridsite.so.$(PATCH_VERSION) \
-                                 $(prefix)/lib/libgridsite.so
-	ln -sf libgridsite.so.$(PATCH_VERSION) \
-                                 $(prefix)/lib/libgridsite.so.$(MAJOR_VERSION)
-	ln -sf libgridsite.so.$(PATCH_VERSION) \
-                                 $(prefix)/lib/libgridsite.so.$(MINOR_VERSION)
-	cp -f  libgridsite_globus.a $(prefix)/lib
-	cp -f  libgridsite_globus.so.$(PATCH_VERSION) $(prefix)/lib
-	ln -sf libgridsite_globus.so.$(PATCH_VERSION) \
-                                 $(prefix)/lib/libgridsite_globus.so
-	ln -sf libgridsite_globus.so.$(PATCH_VERSION) \
-                                 $(prefix)/lib/libgridsite_globus.so.$(MAJOR_VERSION)
-	ln -sf libgridsite_globus.so.$(PATCH_VERSION) \
-                                 $(prefix)/lib/libgridsite_globus.so.$(MINOR_VERSION)
 	cp -f ../CHANGES ../README ../INSTALL ../LICENSE ../VERSION \
                $(prefix)/share/doc/gridsite-$(PATCH_VERSION)
 	cp -f ../doc/*.html ../doc/*.conf ../doc/*.1 ../doc/*.8 ../doc/*.sh \
@@ -284,6 +269,25 @@
 	cp -f gsexec $(prefix)/sbin
 	cp -f mod_gridsite.so $(prefix)/lib/httpd/modules
 
+install-lib:
+	mkdir -p $(prefix)/lib
+	cp -f  libgridsite.a $(prefix)/lib
+	cp -f  libgridsite.so.$(PATCH_VERSION) $(prefix)/lib
+	ln -sf libgridsite.so.$(PATCH_VERSION) \
+                                 $(prefix)/lib/libgridsite.so
+	ln -sf libgridsite.so.$(PATCH_VERSION) \
+                                 $(prefix)/lib/libgridsite.so.$(MAJOR_VERSION)
+	ln -sf libgridsite.so.$(PATCH_VERSION) \
+                                 $(prefix)/lib/libgridsite.so.$(MINOR_VERSION)
+	cp -f  libgridsite_globus.a $(prefix)/lib
+	cp -f  libgridsite_globus.so.$(PATCH_VERSION) $(prefix)/lib
+	ln -sf libgridsite_globus.so.$(PATCH_VERSION) \
+                                 $(prefix)/lib/libgridsite_globus.so
+	ln -sf libgridsite_globus.so.$(PATCH_VERSION) \
+                                 $(prefix)/lib/libgridsite_globus.so.$(MAJOR_VERSION)
+	ln -sf libgridsite_globus.so.$(PATCH_VERSION) \
+                                 $(prefix)/lib/libgridsite_globus.so.$(MINOR_VERSION)
+
 install-slashgrid: slashgrid
 	cp -f slashgrid $(prefix)/sbin
 	cp -f slashgrid.init $(RPM_BUILD_ROOT)/etc/rc.d/init.d/slashgrid
